code/datums/cleaning.dm

```diff
@@ -0,0 +1,81 @@
+//For handling standard click-to-clean items like soap and mops.
+/datum/proc/cleaning_act(atom/target, mob/user, atom/cleaner, cleanspeed = 50, ismop = FALSE)
+	var/innatecleaner = FALSE //If the user cleaning is innately able to clean, i.e. Lusty Xenomorph Maid
+	var/cmag_cleantime = 50 //The cleaning time for cmagged objects is locked to this, for balance reasons
+
+	if(user == cleaner)
+		innatecleaner = TRUE
+
+	if(ismop)
+		if(cleaner.reagents.total_volume < 1)
+			to_chat(user, "<span class='warning'>Your mop is dry!</span>")
+			return
```

Comment:
not a fan of this parameter, I would prefer there be an `obj/item/proc/can_clean(atom/target)` that returns whether or not a soap, mop, etc. can clean